src/wallet/connection.ts

```diff
@@ -88,4 +91,95 @@ class WalletServiceConnection extends BaseConnection {
   }
 }
 
-export default WalletServiceConnection;
+/**
+ * This connection is used when the wallet has websocket disabled.
+ */
+export class DummyWalletServiceConnection extends EventEmitter implements IConnection {
+  public readonly isDummyConnection = true;
+  public walletId?: string;
+
+  private currentNetwork: string;
+  private currentServer: string;
+  protected state: ConnectionState;
+
+  constructor(options: WalletServiceConnectionParams = { walletId: 'dummy-wallet' }) {
+    super();
+
+    const {
+      walletId,
+      network,
+      servers,
+    } = {
+      ...DEFAULT_PARAMS,
+      ...options,
+    };
+
+    if (!network) {
+      throw Error('You must explicitly provide the network.');
+    }
+
+    this.walletId = walletId;
+    this.currentNetwork = network || 'dummynet';
+    this.state = ConnectionState.CLOSED;
+    this.currentServer = servers[0] || config.getServerUrl();
+  }
+
+  start(): void {
+    if (!this.walletId) {
+      throw new Error('Wallet id should be set before connection start.');
+    }
+  }
+
+  stop(): void {
+    this.removeAllListeners();
+    this.setState(ConnectionState.CLOSED);
+  }
+
+  endConnection(): void {
+    // There is no connection to end
+  }
+
+  setup(): void {
+    // There is nothing to setup
+  }
+
+  handleWalletMessage(_wsData: any): void {
+    // There is nothing to handle
+  }
+
+  onConnectionChange(value: boolean): void {
+    if (value) {
+      this.setState(ConnectionState.CONNECTED);
+    } else {
+      this.setState(ConnectionState.CONNECTING);
+    }
+  }
+
+  setState(state: ConnectionState): void {
+    this.state = state;
+  }
+
+  getCurrentServer(): string {
+    return this.currentServer;
+  }
+
+  getCurrentNetwork(): string {
+    return this.currentNetwork;
+  }
+
+  setWalletId(walletId: string) {
+    this.walletId = walletId;
+  }
+
+  isStateClosed(): boolean {
```

Comment:
You don't have this method in the other connections, why do you need it?

Reply:
Yes, this method is not part of the interface. It is here just to make the test cover the` close` method, without exposing the internal state in the class, which would be a violation of the *Open Close Principle*.